src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -604,29 +604,7 @@ class RideHailManager(
           RideHailResponse(request, Some(travelProposal))
         }
       request.customer.personRef ! rideHailResponse
-      val maybePreviousResponse = rideHailResponseCache.get(request.customer)
-      maybePreviousResponse match {
-        case Some(previousRideHailResponse)
-            if previousRideHailResponse.request.requestId < request.requestId &&
-              previousRideHailResponse.request.departAt >= request.departAt =>
-          /* We log an error if an identical inquiry with a time stamp before the previously cached response as illogical
-           * behavior, but we cannot make a stronger claim here that NO previous response is cached. This is because if an
-           * agent makes an inquiry -- and the response is cached here -- but doesn't choose ride hail
-           * as a mode, there is no simple way to remove that cached response inside the RHM. We can still safely
-           * overwrite the response below with the latest.
-           * Sometimes inquiries get to this point not in order they were sent (because of the router with multiple
-           * actors is involved) so we need to check the request ids too
-           */
-          log.error(
-            s"Customer ${request.customer.personId} has made two RideHail Inquiries, with the departAt for the " +
-            s"second being before or equal to the first: (${previousRideHailResponse.request.departAt} < ${request.departAt}. This is " +
-            s"likely to cause logical errors."
-          )
-        case _ =>
-      }
-      if (maybePreviousResponse.isEmpty || maybePreviousResponse.get.request.requestId < request.requestId) {
-        rideHailResponseCache.put(request.customer, rideHailResponse)
-      }
+      rideHailResponseCache.put(request.customer, getActualResponses(request) :+ rideHailResponse)
```

Comment:
Is there still a possible issue with duplicate ridehail inquiries? Should we still log something? I guess since there are two, then we at least expect both to be removed at some point - or should we remove multiple when removing?

Reply:
Now we keep all the actual ridehail inquiries we cannot find logical errors at this point.
We do remove not actual RH inquiries. Method `getActualResponses` filter out outdated responses.